### PR TITLE
CI: pre-commit autoupdate to fix CI

### PR DESCRIPTION
**Pull request by MarcoGorelli**
new click release broke previous version of `black`, see https://github.com/psf/black/issues/2964

### CODE REVIEW

**Review by simonjayhawkins — APPROVED** [review]
Thanks @MarcoGorelli lgtm

OK to merge once Code Checks / pre-commit is green (locally OK so should be no different here)

**Comment by simonjayhawkins** [comment]
I assume will will get the same error on 1.4.x?

**Comment by simonjayhawkins** [comment]
> I assume will will get the same error on 1.4.x?

hmm, merged #46550 a couple of hours ago after the problem was noticed on main. pre-commit was green on 1.4.x. more investigation required.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ repos:
         pass_filenames: true
         require_serial: false
 -   repo: https://github.com/python/black
-    rev: 22.1.0
+    rev: 22.3.0
```
Comment (twoertwein) [inline .pre-commit-config.yaml:19]: also need to update environment.yml and requirements-dev.txt

**Comment by MarcoGorelli** [comment]
I think it's just that on the 1.4.x branch the run used the old cached dependencies:
```
Cache Size: ~63 MB (65887285 B)
```
which would've had the previous `click` version

Don't know how long the cache lasts, but I think this should probably be backported

File: .pre-commit-config.yaml (hunk 1 above)
Comment (MarcoGorelli) [reply .pre-commit-config.yaml:19]: good catch - might need a CI check for this else we'll never remember each time 😄

**Comment by MarcoGorelli** [comment]
> OK to merge once Code Checks / pre-commit is green 

merging then

**Comment by MarcoGorelli** [comment]
@meeseeksdev backport 1.4.x

**Pull request merged**